src/components/ProductListing.jsx

```diff
@@ -0,0 +1,23 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchProducts } from '../redux/products/products';
+import ProductComponent from './ProductComponent';
+
+const ProductPage = () => {
+  const productData = useSelector((state) => state.allProducts.products);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (productData.length === 0) {
+      dispatch(fetchProducts());
+    }
+  }, []);
+
+  return (
+    <div className="products-section">
+      <ProductComponent />
+    </div>
+  );
+};
+
+export default ProductPage;
```

Comment:
- [ ] It is required that the Home page has a list of categories that could be filtered using given parameters, e.g. category name. Kindly implement a search feature so that users can filter categories by name. This will improve your users' experience and interaction.

Reply:
Created a search bar for better user experience and interaction

Reply:
Re-recorded the project presentation